idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/WrapWithSafeLetCall.kt

```diff
@@ -0,0 +1,277 @@
+        // If no symbol in the local scope contains the called symbol, then the symbol must be a member symbol.
+        return scope.scopes.getCallableSymbols { it.identifierOrNullIfSpecial == calleeName }.all { it != propertySymbol }
+    }
+
+    val forUnsafeInfixCall = diagnosticFixFactory<KtFirDiagnostic.UnsafeInfixCall> { diagnostic ->
+        createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(diagnostic.receiverExpression)
+    }
+
+    val forUnsafeOperatorCall = diagnosticFixFactory<KtFirDiagnostic.UnsafeOperatorCall> { diagnostic ->
+        createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(diagnostic.receiverExpression)
+    }
+
+    val forArgumentTypeMismatch = diagnosticFixFactory<KtFirDiagnostic.ArgumentTypeMismatch> { diagnostic ->
+        if (diagnostic.isMismatchDueToNullability) createWrapWithSafeLetCallInputForNullableExpression(diagnostic.psi.wrappingExpressionOrSelf)
+        else emptyList()
+    }
+
+    private fun KtAnalysisSession.createWrapWithSafeLetCallInputForNullableExpressionIfMoreThanImmediateParentIsWrapped(
+        nullableExpression: KtExpression?,
+        isImplicitInvokeCallToMemberProperty: Boolean = false,
+    ): List<IntentionAction> {
+        val surroundingExpression = nullableExpression?.typableParentExpression
+        if (
+            surroundingExpression == null ||
+            // If the surrounding expression is at a place that accepts null value, then we don't provide wrap with let call because the
+            // plain safe call operator (?.) is a better fix.
+            isExpressionAtNullablePosition(surroundingExpression) ||
+            // In addition, if there is no parent that is at a nullable position, then we don't offer wrapping with let either because
+            // it still doesn't fix the code. Hence, the plain safe call operator is a better fix.
+            findParentExpressionAtNullablePosition(nullableExpression) == null
+        ) {
+            return emptyList()
+        }
+        return createWrapWithSafeLetCallInputForNullableExpression(nullableExpression, isImplicitInvokeCallToMemberProperty)
+    }
+
+    private fun KtAnalysisSession.createWrapWithSafeLetCallInputForNullableExpression(
+        nullableExpression: KtExpression?,
+        isImplicitInvokeCallToMemberProperty: Boolean = false,
+        parentExpression: KtExpression? = findParentExpressionAtNullablePosition(nullableExpression)
+            ?: nullableExpression?.typableParentExpression
+    ): List<IntentionAction> {
+        if (nullableExpression == null || parentExpression == null) return emptyList()
+        val existingNames = nullableExpression.containingKtFile.getScopeContextForPosition(nullableExpression).scopes.getCallableNames()
+            .mapNotNull { it.identifierOrNullIfSpecial }
+        val candidateNames = listOfNotNull("it", getDeclaredParameterNameForArgument(nullableExpression))
```

Comment:
Nit: I expect the order matters here, i.e., we prefer "it" to be considered first? If so, `// NB: ...` comment?